module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```
@@ -130,11 +130,102 @@ public function addCss($css)
     public function addJs($js)
     {
         if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+            $this->addJsEntry($js);
+        } elseif (isset($js['file'])) {
+            $this->addJsEntry($js);
+        } elseif (isset($js[0])) {
+            foreach ($js as $current) {
+                $this->addJsEntry($current);
+            }
+        } elseif ($js == []) {
+            return;
+        } else {
+            trigger_error("Invalid JS entry format: " . print_r($js, true));
```

Would it be better to throw an exception rather than trigger an error here?